graph/src/components/versions/registry.rs

```diff
@@ -0,0 +1,87 @@
+use crate::prelude::FeatureFlag;
+use lazy_static::lazy_static;
+use semver::{Version, VersionReq};
+use std::collections::HashMap;
+
+lazy_static! {
+    pub static ref VERSIONS: HashMap<Version, Vec<FeatureFlag>> = {
+        let mut supported_versions: Vec<(Version, Vec<FeatureFlag>)> = vec![
+            // baseline version
+            (Version::new(1, 0, 0), vec![]),
+            // Versions with feature flags
+            (Version::new(1, 1, 0), vec![FeatureFlag::BasicOrdering])
+        ];
+
+        let mut map = HashMap::new();
+
+        // Sort version by major, minor, patch, from higher to lower.
+        supported_versions.sort_by(|a, b| b.0.partial_cmp(&a.0).unwrap());
+
+        for (version, flags) in supported_versions {
+            map.insert(version, flags);
+        }
+
+        map
+    };
+
+    static ref LATEST_VERSION: String = {
+        let keys: Vec<Version> = VERSIONS.clone().into_keys().collect();
+
+        // Versions are sorted by major, minor, patch, from higher to lower.
+        keys.first().unwrap().to_string()
+    };
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ApiVersion {
+    pub version: Version,
+    features: Vec<FeatureFlag>,
+}
+
+impl ApiVersion {
+    pub fn new(version_requirement: &VersionReq) -> Result<Self, String> {
+        let version = Self::resolve(&version_requirement);
```

Comment:
It would be cleaner if `resolve` returned `Result<Self, String>` instead of an `Option`

Reply:
It does now, I did a refactor